### DISCUSSION TITLE
output of `brew audit`

### DISCUSSION THREAD

**Discussion by ipatch** [Homebrew/discussions#1860] [Writing Formulae/Casks] [answered]
been working with `brew audit` and `brew style` for the past couple of days.

it seems like there are some runtime issues, possibly ruby related when i run `brew audit ./some/path/to/formula.rb`

running that above command outputs a ruby stacktrace from what i can see, and not sure if the `audit` cmd is suppose to output or if something is wrong with my `brew audit`

**example**

```
 brew audit --formula $ghforks/homebrew-freecad/Formula/boost@1.75.0.rb
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /opt/code/github/public/forks/homebrew-freecad/Formula/boost@1.75.0.rb:1:0.
To see the complete backtrace run rubocop -d.
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /opt/code/github/public/forks/homebrew-freecad/Formula/boost@1.75.0.rb:1:0.
To see the complete backtrace run rubocop -d.

1 error occurred:
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /opt/code/github/public/forks/homebrew-freecad/Formula/boost@1.75.0.rb:1:0.
undefined method `metadata' for nil:NilClass
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:79:in `display_error_summary'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:58:in `display_summary'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:27:in `block in execute_runner'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command.rb:11:in `run'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/environment.rb:18:in `run'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:65:in `run_command'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:72:in `execute_runners'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:41:in `run'
/usr/local/Homebrew/Library/Homebrew/utils/rubocop.rb:12:in `<main>'
Error: Failure while executing; `/usr/bin/env XDG_CACHE_HOME=/Users/capin/Library/Caches/Homebrew/style /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby /usr/local/Homebrew/Library/Homebrew/utils/rubocop.rb --format json --force-exclusion --parallel --except FormulaAuditStrict --config /usr/local/Homebrew/Library/.rubocop.yml /opt/code/github/public/forks/homebrew-freecad/Formula/boost@1.75.0.rb` exited with 2. Here's the output:
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /opt/code/github/public/forks/homebrew-freecad/Formula/boost@1.75.0.rb:1:0.
To see the complete backtrace run rubocop -d.
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /opt/code/github/public/forks/homebrew-freecad/Formula/boost@1.75.0.rb:1:0.
To see the complete backtrace run rubocop -d.

1 error occurred:
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /opt/code/github/public/forks/homebrew-freecad/Formula/boost@1.75.0.rb:1:0.
undefined method `metadata' for nil:NilClass
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:79:in `display_error_summary'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:58:in `display_summary'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:27:in `block in execute_runner'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command.rb:11:in `run'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/environment.rb:18:in `run'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:65:in `run_command'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:72:in `execute_runners'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:41:in `run'
/usr/local/Homebrew/Library/Homebrew/utils/rubocop.rb:12:in `<main>'
{"metadata":{"rubocop_version":"1.18.3","ruby_engine":"ruby","ruby_version":"2.6.3","ruby_patchlevel":"62","ruby_platform":"x86_64-darwin14"},"files":[{"path":"/opt/code/github/public/forks/homebrew-freecad/Formula/boost@1.75.0.rb","offenses":[]}],"summary":{"offense_count":0,"target_file_count":1,"inspected_file_count":1}}
```

---

**TL;DR**

> Yeah, I've seen a number of third-party formulae that resort to version-pinning dependencies in the same tap.
> 
> @ipatch, I suddenly realized that there's a shorthand for "dependency in same tap as formula":
> 
> ```
>   depends_on "./icu4c@67.1"
> ```
> 
> This works for both `install` and `audit`, as far as I can tell.

**Comment by gromgit** [comment, marked as answer]
Do you get the same errors if you audit the formula in a proper tap?
```
brew tap-new ipatch/tap
cp $ghforks/homebrew-freecad/Formula/boost@1.75.0.rb $(brew --repo ipatch/tap)/Formula/
brew audit ipatch/tap/boost@1.75.0
```

**Reply by ipatch** [reply]
please forgive right side prompt messages in the below copy paste. i need to set a key binding in my terminal to disable the prompt doing such things, just haven't gotten to it yet.

```
╰─λ brew tap-new ipatch/footap                                                                                                               0 (0.001s)
Initialized empty Git repository in /usr/local/Homebrew/Library/Taps/ipatch/homebrew-footap/.git/
[main (root-commit) b2b7b68] Create ipatch/footap tap
 3 files changed, 88 insertions(+)
 create mode 100644 .github/workflows/publish.yml
 create mode 100644 .github/workflows/tests.yml
 create mode 100644 README.md
==> Created ipatch/footap
/usr/local/Homebrew/Library/Taps/ipatch/homebrew-footap

When a pull request making changes to a formula (or formulae) becomes green
(all checks passed), then you can publish the built bottles.
To do so, label your PR as `pr-pull` and the workflow will be triggered.
╭─capin at rogue in /Users/capin/ltmp
╰─λ cp -iv $ghforks/homebrew-freecad/Formula/boost@1.75.0.rb (brew --repo ipatch/footap)/Formula/                                            0 (2.381s)
/opt/code/github/public/forks/homebrew-freecad/Formula/boost@1.75.0.rb -> /usr/local/Homebrew/Library/Taps/ipatch/homebrew-footap/Formula/boost@1.75.0.rb
╭─capin at rogue in /Users/capin/ltmp
╰─λ brew audit ipatch/footap/boost@1.75                                                                                                      0 (1.000s)
Error: No available formula or cask with the name "ipatch/footap/boost@1.75". Did you mean ipatch/footap/boost@1.75.0?
╭─capin at rogue in /Users/capin/ltmp
╰─λ brew audit ipatch/footap/boost@1.75.0                                                                                                    0 (1.368s)
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /usr/local/Homebrew/Library/Taps/ipatch/homebrew-footap/Formula/boost@1.75.0.rb:1:0.
To see the complete backtrace run rubocop -d.
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /usr/local/Homebrew/Library/Taps/ipatch/homebrew-footap/Formula/boost@1.75.0.rb:1:0.
To see the complete backtrace run rubocop -d.

1 error occurred:
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /usr/local/Homebrew/Library/Taps/ipatch/homebrew-footap/Formula/boost@1.75.0.rb:1:0.
undefined method `metadata' for nil:NilClass
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:79:in `display_error_summary'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:58:in `display_summary'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:27:in `block in execute_runner'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command.rb:11:in `run'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/environment.rb:18:in `run'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:65:in `run_command'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:72:in `execute_runners'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:41:in `run'
/usr/local/Homebrew/Library/Homebrew/utils/rubocop.rb:12:in `<main>'
Error: Failure while executing; `/usr/bin/env XDG_CACHE_HOME=/Users/capin/Library/Caches/Homebrew/style /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby /usr/local/Homebrew/Library/Homebrew/utils/rubocop.rb --format json --force-exclusion --parallel --except FormulaAuditStrict --config /usr/local/Homebrew/Library/.rubocop.yml /usr/local/Homebrew/Library/Taps/ipatch/homebrew-footap/Formula/boost@1.75.0.rb` exited with 2. Here's the output:
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /usr/local/Homebrew/Library/Taps/ipatch/homebrew-footap/Formula/boost@1.75.0.rb:1:0.
To see the complete backtrace run rubocop -d.
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /usr/local/Homebrew/Library/Taps/ipatch/homebrew-footap/Formula/boost@1.75.0.rb:1:0.
To see the complete backtrace run rubocop -d.

1 error occurred:
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /usr/local/Homebrew/Library/Taps/ipatch/homebrew-footap/Formula/boost@1.75.0.rb:1:0.
undefined method `metadata' for nil:NilClass
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:79:in `display_error_summary'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:58:in `display_summary'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:27:in `block in execute_runner'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command.rb:11:in `run'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/environment.rb:18:in `run'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:65:in `run_command'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:72:in `execute_runners'
/usr/local/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:41:in `run'
/usr/local/Homebrew/Library/Homebrew/utils/rubocop.rb:12:in `<main>'
{"metadata":{"rubocop_version":"1.18.3","ruby_engine":"ruby","ruby_version":"2.6.3","ruby_patchlevel":"62","ruby_platform":"x86_64-darwin14"},"files":[{"path":"/usr/local/Homebrew/Library/Taps/ipatch/homebrew-footap/Formula/boost@1.75.0.rb","offenses":[]}],"summary":{"offense_count":0,"target_file_count":1,"inspected_file_count":1}}
```

also you can find the formula file i'm working with below

https://github.com/ipatch/homebrew-freecad

feel free to test on your end and see if you get the same issues.

**Reply by gromgit** [reply]
Hmm...
```
$ brew tap ipatch/freecad
==> Tapping ipatch/freecad
Cloning into '/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad'...
remote: Enumerating objects: 1723, done.
remote: Counting objects: 100% (479/479), done.
remote: Compressing objects: 100% (277/277), done.
remote: Total 1723 (delta 192), reused 417 (delta 151), pack-reused 1244
Receiving objects: 100% (1723/1723), 409.86 KiB | 6.12 MiB/s, done.
Resolving deltas: 100% (873/873), done.
Tapped 25 formulae (44 files, 627.7KB).

$ brew audit ipatch/freecad/boost@1.75.0 
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb:1:0.
To see the complete backtrace run rubocop -d.
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb:1:0.
To see the complete backtrace run rubocop -d.

1 error occurred:
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb:1:0.
undefined method `metadata' for nil:NilClass
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:79:in `display_error_summary'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:58:in `display_summary'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:27:in `block in execute_runner'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command.rb:11:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/environment.rb:18:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:65:in `run_command'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:72:in `execute_runners'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:41:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils/rubocop.rb:12:in `<main>'
Error: Failure while executing; `/usr/bin/env XDG_CACHE_HOME=/home/aho/.cache/Homebrew/style /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils/rubocop.rb --format json --force-exclusion --parallel --except FormulaAuditStrict --config /home/linuxbrew/.linuxbrew/Homebrew/Library/.rubocop.yml /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb` exited with 2. Here's the output:
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb:1:0.
To see the complete backtrace run rubocop -d.
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb:1:0.
To see the complete backtrace run rubocop -d.

1 error occurred:
An error occurred while FormulaAudit/DependencyOrder cop was inspecting /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb:1:0.
undefined method `metadata' for nil:NilClass
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:79:in `display_error_summary'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:58:in `display_summary'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:27:in `block in execute_runner'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/command.rb:11:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli/environment.rb:18:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:65:in `run_command'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:72:in `execute_runners'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle/ruby/2.6.0/gems/rubocop-1.18.3/lib/rubocop/cli.rb:41:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils/rubocop.rb:12:in `<main>'
{"metadata":{"rubocop_version":"1.18.3","ruby_engine":"ruby","ruby_version":"2.6.3","ruby_patchlevel":"62","ruby_platform":"x86_64-linux"},"files":[{"path":"/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb","offenses":[]}],"summary":{"offense_count":0,"target_file_count":1,"inspected_file_count":1}}
```
Turns out the problem is in your formula, specifically:
```
  depends_on "#{@tap}/icu4c@67.1"
```
I can't recall offhand an expansion for "this formula's tap", so you'll likely have to hardcode the tap name. If the formulation you used works as-is, you're relying on an implementation detail that may change at any time.

**Reply by carlocab** [reply]
The expansion is just `tap`. If you want to use `icu4c@67.1` from the current tap, do
```ruby
depends_on "#{tap}/icu4c@67.1"
```

**Reply by gromgit** [reply]
That doesn't seem to work:
```
$ brew audit ipatch/freecad/boost@1.75.0
Error: no block given
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb:22:in `tap'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb:22:in `<class:BoostAT1750>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb:1:in `block in load_formula'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:78:in `module_eval'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:78:in `block in load_formula'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:90:in `load_formula'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:110:in `load_formula_from_path'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:195:in `load_file'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:350:in `load_file'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:185:in `klass'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:180:in `get_formula'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:340:in `get_formula'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:412:in `factory'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cli/named_args.rb:104:in `load_formula_or_cask'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cli/named_args.rb:59:in `block in to_formulae_and_casks'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cli/named_args.rb:58:in `each'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cli/named_args.rb:58:in `flat_map'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cli/named_args.rb:58:in `to_formulae_and_casks'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dev-cmd/audit.rb:126:in `audit'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.rb:122:in `<main>'

$ brew install ipatch/freecad/boost@1.75.0
Error: no block given
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb:22:in `tap'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb:22:in `<class:BoostAT1750>'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/ipatch/homebrew-freecad/Formula/boost@1.75.0.rb:1:in `block in load_formula'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:78:in `module_eval'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:78:in `block in load_formula'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:90:in `load_formula'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:110:in `load_formula_from_path'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:195:in `load_file'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:350:in `load_file'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:185:in `klass'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:180:in `get_formula'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:340:in `get_formula'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formulary.rb:412:in `factory'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cli/parser.rb:636:in `block in formulae'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cli/parser.rb:632:in `map'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cli/parser.rb:632:in `formulae'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cli/parser.rb:311:in `parse'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/cmd/install.rb:131:in `install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.rb:122:in `<main>'
```

**Reply by carlocab** [reply]
Weird. In `brew irb`, `"#{Formula["foo"].tap}/foo"` seems to be be interpolated into the right thing, so I'd have thought it worked here. Maybe `depends_on "#{self.tap}/icu4c@67.1"`? In any case, it's probably just simpler to hard-code the tap...

**Reply by gromgit** [reply]
It turns out `#{@tap}` expands fine during `brew install`, just not in a `brew audit`. My Ruby is nonexistent, but perhaps it's because `install` creates a Formula instance, while `audit` doesn't?

In any case, if my Google-fu hasn't failed me:
```
irb(main):007:0> p Formula["hello"].method(:tap)
#<Method: Formulary::FormulaNamespace16706be1e9330b9046fe91cf7121cb80::Hello(Formula)#tap>
=> #<Method: Formulary::FormulaNamespace16706be1e9330b9046fe91cf7121cb80::Hello(Formula)#tap>
```
which seems to be declared as a private API? https://rubydoc.brew.sh/Formulary/TapLoader.html#tap-instance_method

**Reply by carlocab** [reply]
Ah, yes, I was suspecting it was a private API. We don't want to rely on that. It should be better to hardcode the tap. Or maybe just use an unqualified `depends_on "icu4c@67.1"` -- unless one really needs non-standard customisations specific to the `icu4c@67.1` in a particular tap beyond pinning it to a specific version. (I imagine it's the version pinning that's useful here, as ICU version bumps are a pain.)

**Reply by gromgit** [reply]
Yeah, I've seen a number of third-party formulae that resort to version-pinning dependencies in the same tap.

@ipatch, I suddenly realized that there's a shorthand for "dependency in same tap as formula":
```
  depends_on "./icu4c@67.1"
```
This works for both `install` and `audit`, as far as I can tell.

**Reply by ipatch** [reply]
```
Error: Could not get runtime dependencies from /usr/local/Homebrew/Library/Taps/freecad/homebrew-freecad/Formula/boost@1.75.0.rb!
Error: No available formula with the name "./icu4c@67.1.rb". Did you mean icu4c@67.1?
╭─capin at rogue in /opt/code/github/public/forks/homebrew-freecad (ipatch.exp.with.github.actions ✚1⚑9)
╰─λ brew audit freecad/freecad/boost@1.75.0                                                                                               0 (01:01.650)
Error: Could not get runtime dependencies from /usr/local/Homebrew/Library/Taps/freecad/homebrew-freecad/Formula/boost@1.75.0.rb!
Error: No available formula with the name "./icu4c@67.1.rb". Did you mean icu4c@67.1?
╭─capin at rogue in /opt/code/github/public/forks/homebrew-freecad (ipatch.exp.with.github.actions ✚1⚑9)
╰─λ brew audit Formula/boost@1.75.0.rb                                                                                                      0 (17.877s)
Error: Failed to load cask: Formula/boost@1.75.0.rb
Cask 'boost@1.75.0' is unreadable: wrong constant name #<Class:0x00007fa57ab88770>
Warning: Treating Formula/boost@1.75.0.rb as a formula.
Error: Could not get runtime dependencies from /usr/local/Homebrew/Library/Taps/freecad/homebrew-freecad/Formula/boost@1.75.0.rb!
Error: No available formula with the name "./icu4c@67.1.rb". Did you mean icu4c@67.1?
```

**Reply by ipatch** [reply]
whoops vim autocomplete got the best of me on that one. 😣

**Reply by ipatch** [reply]
@gromgit thanks so much for helping out with this too 🙏
